RetroAchievements/RAWeb
# feat(game): add script for migrating release dates to actual timestamps

This PR adds a new Artisan command & action for migrating `GameData.Released` to `GameData.released_at` and `GameData.released_at_granularity`.

`released_at_granularity` is a new nullable column which can contain "day" (YYYY-MM-DD), "month" (YYYY-MM), or "year" (YYYY).

**How to test this PR:**
After running `sail artisan migrate`,
```
sail artisan ra:platform:game:convert-released-to-timestamp 1 # convert a single game (target by game id)
sail artisan ra:platform:game:convert-released-to-timestamp # convert all games
```

Next steps:
* Either migrate the "Released:" field on the game page devbox to Filament, or in the short-term, have it double write to these columns and use a datepicker control to ensure structured data.
* Change the few queries we have which point to `Released` to instead point to `released_at` and possibly `released_at_granularity`.
* Drop `GameData.Released`.

This PR solely focuses on making sure we have this initial command & action correct.
